src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java

```diff
@@ -0,0 +1,323 @@
+        }
+
+        tutorialGroupService.registerStudent(userToRegister, tutorialGroupFromDatabase, TutorialGroupRegistrationType.INSTRUCTOR_REGISTRATION);
+        return ResponseEntity.noContent().build();
+    }
+
+    /**
+     * POST /courses/:courseId/tutorial-groups/:tutorialGroupId/register-multiple" : Register multiple users to the tutorial group
+     *
+     * @param courseId        the id of the course to which the tutorial group belongs to
+     * @param tutorialGroupId the id of the tutorial group to which the users should be registered to
+     * @param studentDtos     the list of students who should be registered to the tutorial group
+     * @return the list of students who could not be registered for the tutorial group, because they could NOT be found in the Artemis database as students of the tutorial group course
+     */
+    @PostMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}/register-multiple")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<Set<StudentDTO>> registerMultipleStudentsToTutorialGroup(@PathVariable Long courseId, @PathVariable Long tutorialGroupId,
+            @RequestBody Set<StudentDTO> studentDtos) {
+        log.debug("REST request to register {} to tutorial group {}", studentDtos, tutorialGroupId);
+        var tutorialGroupFromDatabase = this.tutorialGroupRepository.findByIdElseThrow(tutorialGroupId);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, tutorialGroupFromDatabase.getCourse(), null);
+        checkEntityIdMatchesPathIds(tutorialGroupFromDatabase, Optional.of(courseId), Optional.of(tutorialGroupId));
+        Set<StudentDTO> notFoundStudentDtos = tutorialGroupService.registerMultipleStudents(tutorialGroupFromDatabase, studentDtos,
+                TutorialGroupRegistrationType.INSTRUCTOR_REGISTRATION);
+        return ResponseEntity.ok().body(notFoundStudentDtos);
+    }
+
+    private void trimStringFields(TutorialGroup tutorialGroup) {
+        if (tutorialGroup.getTitle() != null) {
+            tutorialGroup.setTitle(tutorialGroup.getTitle().trim());
+        }
+        if (tutorialGroup.getAdditionalInformation() != null) {
+            tutorialGroup.setAdditionalInformation(tutorialGroup.getAdditionalInformation().trim());
+        }
+        if (tutorialGroup.getCampus() != null) {
+            tutorialGroup.setCampus(tutorialGroup.getCampus().trim());
+        }
+    }
+
+    private void checkTitleIsUnique(TutorialGroup tutorialGroup) {
+        if (tutorialGroupRepository.existsByTitleAndCourse(tutorialGroup.getTitle(), tutorialGroup.getCourse())) {
+            throw new BadRequestException("A tutorial group with this title already exists in the course.");
+        }
+    }
+
+    private static void overrideValues(TutorialGroup sourceTutorialGroup, TutorialGroup originalTutorialGroup) {
+        originalTutorialGroup.setTitle(sourceTutorialGroup.getTitle());
```

Comment:
Make those setter return `Tutorgroup` so you can write
```java
originalTutorialGroup
  .setTeachingAssisteant(...)
  .setAdditionalInformation(...)...
```

Reply:
I would rather not do this. I read up on it online and apparently this causes problems as it fails the JavaBean Specification. 
```
it violates the JavaBean spec, so it will break with various tools and libraries, especially JSP builders and Spring.
```
See:  https://stackoverflow.com/questions/1345001/is-it-bad-practice-to-make-a-setter-return-this